tests/pipeline/test_python_component.py

```diff
@@ -1,21 +1,62 @@
+import textwrap
+
 import dask.dataframe as dd
 import pandas as pd
 import pyarrow as pa
 from fondant.component import DaskLoadComponent, PandasTransformComponent
 from fondant.pipeline import Pipeline, lightweight_component
 
 
-def test_lightweight_component():
-    pipeline = Pipeline(name="dummy-pipeline", base_path="./data")
+def test_build_python_script():
+    @lightweight_component()
+    class CreateData(DaskLoadComponent):
+        def load(self) -> dd.DataFrame:
+            df = pd.DataFrame(
+                {
+                    "x": [1, 2, 3],
+                    "y": [4, 5, 6],
+                },
+                index=pd.Index(["a", "b", "c"], name="id"),
+            )
+            return dd.from_pandas(df, npartitions=1)
+
+    assert CreateData.image().script == textwrap.dedent(
+        """\
+        from typing import *
+        import typing as t
+
+        import dask.dataframe as dd
+        import fondant
+        import pandas as pd
+        from fondant.component import *
+        from fondant.core import *
+
+
+        class CreateData(DaskLoadComponent):
+            def load(self) -> dd.DataFrame:
+                df = pd.DataFrame(
+                    {
+                        "x": [1, 2, 3],
+                        "y": [4, 5, 6],
+                    },
+                    index=pd.Index(["a", "b", "c"], name="id"),
+                )
+                return dd.from_pandas(df, npartitions=1)
+    """,
+    )
+
+
+def test_lightweight_component(tmp_path_factory):
+    pipeline = Pipeline(
+        name="dummy-pipeline",
+        base_path="/home/robbe/workspace/fondant/tests/pipeline/data",
```

Comment:
base path ?

Reply:
Good catch. Had some issues with the relative path. Will check if I still run into them.